src/Data/Conduit/Tar.hs

```diff
@@ -820,6 +845,157 @@ filePathConduit = do
         Nothing -> return ()
 
 
+-- | A smart tarball creator, that can accept files and folders alike, while trying its best to
+-- preserve proper tar structure by adding missing directory structures.
+--
+-- ==== __Examples__
+--
+-- If executed inside @"tar-conduit"@ package folder, this will produce a tar with all this files
+-- and folders:
+--
+-- >>> runConduitRes (yield "src/Data" .| tarFiles Nothing Nothing (Just "src") .| tar)
+--
+-- @
+-- Data\/
+-- Data\/Conduit\/
+-- Data\/Conduit\/Tar\/
+-- Data\/Conduit\/Tar\/Windows.hs
+-- Data\/Conduit\/Tar\/Types.hs
+-- Data\/Conduit\/Tar\/Unix.hs
+-- Data\/Conduit\/Tar.hs
+-- @
+--
+-- While this will add a prefix to all included files and folders:
+--
+-- >>> prefix <- makeDirectory "foo/bar"
+-- >>> runConduitRes (yield "src/Data/Conduit/Tar" .| tarFiles (Just prefix) Nothing (Just "src/Data/Conduit") .| tar)
+--
+-- @
+-- foo\/
+-- foo\/bar\/
+-- foo\/bar\/Tar\/
+-- foo\/bar\/Tar\/Windows.hs
+-- foo\/bar\/Tar\/Types.hs
+-- foo\/bar\/Tar\/Unix.hs
+-- @
+--
+-- There is also a way to control how deep should directories be traversed:
+--
+-- >>> runConduitRes (yield "src/Data" .| tarFiles Nothing (Just 3) (Just "src/Data") .| tar)
+--
+-- @
+-- Conduit\/
+-- Conduit\/Tar\/
+-- Conduit\/Tar.hs
+-- @
+--
+-- @since 0.3.3
+tarFiles ::
+       (MonadThrow m, MonadResource m)
+    => Maybe FileInfo -- ^ Base directory where all of the files will be placed in. If `Nothing`
```

Comment:
It may seem like overkill, but I'd probably go for a settings type here instead of a list of arguments. It can have clear names and allows for future extensibility.